pkg/iss/config.go

```diff
@@ -88,6 +88,13 @@ type Config struct {
 	// Must not be negative.
 	MsgBufCapacity int
 
+	// Number of most recent epochs that are older than the latest stable checkpoint.
+	RetainedEpochs int
```

Comment:
Can you think of a good reason to let a node fall for more than one epoch benind?

Reply:
To avoid false-positives in detecting nodes that fall behind. I expect this parameter to end up being 1, but it might be interesting to experiment with higher values at some point too.

Reply:
Is it because we artificially chose to create checkpoints at the same pace as we advance through ISS epochs?

Reply:
Not necessarily. Even temporary glitches in a node's performance or network connectivity might make it seem to have fallen behind.

But my main reasoning behind having this parameter is that it costs almost nothing to be able to tune this value instead of having a hard-coded 1. So the very worst case is that it will always be set to 1. And if we do want to try a different value, we have a parameter for it.

Reply:
> temporary glitches in a node's performance or network connectivity

... is exactly the case to treat the node as fallen behind, no?

Since the main reason for checkpoints is to perform garbage-collection of protocol history, it doesn't seem particularly useful to keep more of it than up to one stable checkpoint behind.

Reply:
> it costs almost nothing to be able to tune this value

Too many configuration parameters may be confusing for the user.

Reply:
The node with the checkpoint can also experience network connectivity glitches, making it falsely suspect another node to have fallen behind. And as I said, it is very well possible that the parameter will end up being 1 most of the time.

> Too many configuration parameters may be confusing for the user.

I don't think this is a big issue, if the parameter has its default value and the user doesn't ever have to touch it.

Anyway, if you feel strongly about this and want to have a hard-coded 1 instead of the parameter, go ahead and push the change directly to this branch (GH should let you do that), it's fine with me, but pls let's move on in this issue.

Reply:
Let it be)